TRLevelControlTests/Base/TestBase.cs

```diff
@@ -0,0 +1,263 @@
+
+    public static TR3Level GetTR3AltTestLevel()
+        => GetTR3Level("TEST2.TR2");
+
+    public static TR4Level GetTR4TestLevel()
+        => GetTR4Level("TEST1.TR4");
+
+    public static TR4Level GetTR4AltTestLevel()
+        => GetTR4Level("TEST2.TR4");
+
+    public static TR5Level GetTR5TestLevel()
+        => GetTR5Level("TEST1.TRC");
+
+    public static TR5Level GetTR5AltTestLevel()
+        => GetTR5Level("TEST2.TRC");
+
+    public static TRLevel GetTR1Level(string level)
+    {
+        TR1LevelReader control = new();
+        return control.ReadLevel(GetReadPath(level, TRGameVersion.TR1));
+    }
+
+    public static TR2Level GetTR2Level(string level)
+    {
+        TR2LevelReader control = new();
+        return control.ReadLevel(GetReadPath(level, TRGameVersion.TR2));
+    }
+
+    public static TR3Level GetTR3Level(string level)
+    {
+        TR3LevelReader control = new();
+        return control.ReadLevel(GetReadPath(level, TRGameVersion.TR3));
+    }
+
+    public static TR4Level GetTR4Level(string level)
+    {
+        TR4LevelReader control = new();
+        return control.ReadLevel(GetReadPath(level, TRGameVersion.TR4));
+    }
+
+    public static TR5Level GetTR5Level(string level)
+    {
+        TR5LevelReader control = new();
+        return control.ReadLevel(GetReadPath(level, TRGameVersion.TR5));
+    }
+
+    public static void ReadWriteTR1Level(string levelName)
+    {
+        TR1LevelReader reader = new();
+        TR1LevelWriter writer = new();
+
+        string pathI = GetReadPath(levelName, TRGameVersion.TR1);
+        string pathO = GetWritePath(levelName, TRGameVersion.TR1);
+
+        TRLevel level = reader.ReadLevel(pathI);
+        writer.WriteLevelToFile(level, pathO);
+
+        byte[] b1 = File.ReadAllBytes(pathI);
+        byte[] b2 = File.ReadAllBytes(pathO);
+
+        CollectionAssert.AreEqual(b1, b2);
+    }
+
+    public static void ReadWriteTR2Level(string levelName)
+    {
+        TR2LevelReader reader = new();
+        TR2LevelWriter writer = new();
+
+        string pathI = GetReadPath(levelName, TRGameVersion.TR2);
+        string pathO = GetWritePath(levelName, TRGameVersion.TR2);
+
+        TR2Level level = reader.ReadLevel(pathI);
+        writer.WriteLevelToFile(level, pathO);
+
+        byte[] b1 = File.ReadAllBytes(pathI);
+        byte[] b2 = File.ReadAllBytes(pathO);
+
+        CollectionAssert.AreEqual(b1, b2);
+    }
+
+    public static void ReadWriteTR3Level(string levelName)
+    {
+        TR3LevelReader reader = new();
+        TR3LevelWriter writer = new();
+
+        string pathI = GetReadPath(levelName, TRGameVersion.TR3);
+        string pathO = GetWritePath(levelName, TRGameVersion.TR3);
+
+        TR3Level level = reader.ReadLevel(pathI);
+        writer.WriteLevelToFile(level, pathO);
+
+        byte[] b1 = File.ReadAllBytes(pathI);
+        byte[] b2 = File.ReadAllBytes(pathO);
+
+        CollectionAssert.AreEqual(b1, b2);
+    }
```

Comment:
There's a lot of repetition here, is there a way to use the game version enum and make the tests more concise?

Reply:
Good idea, thanks. I've streamlined these (1-3 only).